src/PrestaShopBundle/DataCollector/HookRegistry.php

```diff
@@ -0,0 +1,168 @@
+     * @param $hookArguments array
+     * @param $file string filepath where the "Hook::exec" call have been done.
+     * @param $line string position in file where the "Hook::exec" call have been done.
+     */
+    public function selectHook($hookName, $hookArguments, $file, $line)
+    {
+        $this->currentHook = array(
+            'name' => $hookName,
+            'args' => $hookArguments,
+            'location' => "$file:$line",
+            'status' => self::HOOK_NOT_CALLED,
+            'modules' => array(),
+        );
+    }
+
+    /**
+     * Notify the registry that the selected hook have been called.
+     */
+    public function hookWasCalled()
+    {
+        $this->currentHook['status'] = self::HOOK_CALLED;
+    }
+
+    /**
+     * @param ModuleInterface $module
+     */
+    public function hookedByModule(ModuleInterface $module)
+    {
+        $this->currentHook['modules'][$module->name] = array(
+            'callback' => array(),
+            'widget' => array(),
+        );
+    }
+
+    /**
+     * A callback have been executed by the module during the Hook dispatch.
+     *
+     * @param ModuleInterface $module
+     * @param $args array All arguments passed to the Module callback.
+     */
+    public function hookedByCallback(ModuleInterface $module, $args)
+    {
+        $this->currentHook['modules'][$module->name]['callback'] = array(
+            'args' => $args,
+        );
+    }
+
+    /**
+     * A widget have been rendered by the module during the Hook dispatch.
+     *
+     * @param ModuleInterface $module
+     * @param $args array All arguments passed to the Module callback.
+     */
+    public function hookedByWidget(ModuleInterface $module, $args)
+    {
+        $this->currentHook['modules'][$module->name]['widget'] = array(
+            'args' => $args,
+        );
+    }
+
+    /**
+     * @return array the list of called hooks.
+     */
+    public function getCalledHooks()
+    {
+        return $this->hooks['called'];
+    }
+
+    /**
+     * @return array the list of uncalled hooks.
+     */
+    public function getNotCalledHooks()
+    {
+        return $this->hooks['notCalled'];
+    }
+
+    /**
+     * @return array the list of dispatched hooks.
+     */
+    public function getHooks()
+    {
+        return $this->hooks['called'] + $this->hooks['notCalled'];
+    }
+
+    /**
+     * Persist the selected hook into the list.
+     *
+     * Theses hooks will be used by the HookDataCollector
+     */
+    public function collect()
+    {
+        $name = $this->currentHook['name'];
+        $status = $this->currentHook['status'];
+
+        $hook = array(
+            'args' => $this->currentHook['args'],
+            'name' => $name,
+            'location' => $this->currentHook['location'],
+            'modules' => $this->currentHook['modules'],
+        );
+
+        if (array_key_exists($name, $this->hooks[$status])) {
+            $this->hooks[$status][$name][] = $hook;
```

Comment:
Did you mean 
```php
$this->hooks[$status][$name] = array();
```

?

Reply:
xD I've refactored this part so much that the condition is even useless! I remove it :+1: